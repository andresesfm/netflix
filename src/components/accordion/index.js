import React, { createContext, useContext, useState } from 'react';
import { Container, Title, Inner, Frame, Header, Body, Item } from './styles/accordion'
import OptForm from "../opt-form";
import faqData from "../../fixtures/faqs";

const ToggleContext = createContext()

const Accordion = ({ children, ...restProps }) => {
    return (<Container>
        <Inner><Title>Frequently Asked Questions</Title>
            <Frame>
                {faqData.map(({ id, header, body }) => (
                    <AItem key={id}>
                        <AHeader>{header}</AHeader>
                        <ABody>{body}</ABody>
                    </AItem>
                ))}
            </Frame>

            <OptForm>
                <OptForm.Input placeholder="Email address" />
                <OptForm.Button>Try it now</OptForm.Button>
                <OptForm.Break />
                <OptForm.Text>Ready to watch? Enter your email to create or restart your membership.</OptForm.Text>
            </OptForm></Inner>
    </Container>);
};

const AItem = ({ children, ...restProps }) => {
    const [toggleShow, setToggleShow] = useState(false)
    return (<ToggleContext.Provider value={{ toggleShow, setToggleShow }}><Item {...restProps}>{children}</Item></ToggleContext.Provider>)
}

const AHeader = ({ children, ...restProps }) => {
    const { toggleShow, setToggleShow } = useContext(ToggleContext)
    return (
        <Header onClick={() => setToggleShow((show) => !show)} {...restProps}>
            {children}
            {toggleShow ?
                (<img src='/images/icons/close-slim.png' alt='Close' />) :
                (<img src='/images/icons/add.png' alt='Open' />)}</Header>
    )
}

const ABody = ({ children, ...restProps }) => {
    const { toggleShow } = useContext(ToggleContext)
    return toggleShow ? <Body {...restProps}>{children}</Body> : null
}


export default Accordion;
