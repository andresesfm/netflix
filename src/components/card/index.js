import React, { createContext, useState, useEffect, useContext } from 'react';
import {
    Container,
    Group,
    Title,
    Subtitle,
    Text,
    Feature,
    FeatureTitle,
    FeatureText,
    FeatureClose,
    Maturity,
    Content,
    Meta,
    Entities,
    Item,
    Image
} from './styles/card'
import { Player } from '../'
export const FeatureContext = createContext()

const Card = ({ children, category, slideRows, ...restProps }) => {
    const [showFeature, setShowFeature] = useState(false);
    const [itemFeature, setItemFeature] = useState({});
    const [selectedGroup, setSelectedGroup] = useState(null);
    return <FeatureContext.Provider value={{ showFeature, setShowFeature, itemFeature, setItemFeature, setSelectedGroup, selectedGroup }}>

        <Group flexDirection='column'  {...restProps}>
            {slideRows.map((slideItem) => (
                <Container key={`${category}-${slideItem.title.toLowerCase()}`} >
                    <Title>{slideItem.title}</Title>
                    <Entities>
                        {slideItem.data.map((item) => (
                            <CItem key={item.docId} item={item} group={slideItem.title}>
                                <Image src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`} />
                                <Meta>
                                    <Subtitle>{item.title}</Subtitle>
                                    <Text>{item.description}</Text>
                                </Meta>
                            </CItem>
                        ))}
                    </Entities>
                    <CFeature category={category} group={slideItem.title}>
                        <Player src='/videos/bunny.mp4' />
                    </CFeature>
                </Container>
            ))}
        </Group>
    </FeatureContext.Provider>;
};

const CItem = ({ children, item, group, ...restProps }) => {
    const { setShowFeature, setItemFeature, setSelectedGroup } = useContext(FeatureContext);
    return <Item
        onClick={() => {
            setItemFeature(item);
            setShowFeature(true);
            setSelectedGroup(group)
        }}
        {...restProps}>
        {children}
    </Item>;
};

const CFeature = ({ children, category, group, ...restProps }) => {
    const { showFeature, itemFeature, setShowFeature, selectedGroup } = useContext(FeatureContext);
    return (showFeature && group === selectedGroup ? <Feature
        src={`images/${category}/${itemFeature.genre}/${itemFeature.slug}/large.jpg`}
        {...restProps}>
        <Content>
            <FeatureTitle>{itemFeature.title}</FeatureTitle>
            <FeatureText>{itemFeature.description}</FeatureText>
            <FeatureClose onClick={() => { setShowFeature(false) }}>
                <img src='/images/icons/close.png' alt='Close' />
            </FeatureClose>
            <Group margin="80px 0" flexDirection="column" alignItems='center'>
                <Maturity rating={itemFeature.maturity}>
                    {itemFeature.maturity < 12 ? 'PG' : itemFeature.maturity}
                </Maturity>
                <FeatureText fontWeight="bold">
                    {itemFeature.genre.charAt(0).toUpperCase() + itemFeature.genre.slice(1)}
                </FeatureText>
            </Group>
        </Content>
        {children}
    </Feature> : null);
};



export default Card;
