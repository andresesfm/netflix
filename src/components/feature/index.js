import React from 'react';
import { Container, Title, SubTitle } from './styles/feature'
import OptForm from '../opt-form';
const Feature = () => {
    return <Container>
        <Title>Unlimited films, TV progams and more.</Title>
        <SubTitle >Watch anywhere. Cancel any time.</SubTitle>
        <OptForm>
            <OptForm.Input placeholder="Email address" />
            <OptForm.Button>Try it now</OptForm.Button>
            <OptForm.Break />
            <OptForm.Text>Ready to watch? Enter your email to create or restart your membership.</OptForm.Text>
        </OptForm>
    </Container>;
};

export default Feature;
