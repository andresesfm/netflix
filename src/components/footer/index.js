import React from 'react';
import { Container, Row, Column, Link, Title, Text, Break } from './styles/footer';


const Footer = () => {
    return <Container>
        <Title>Questions? Contact us.</Title>
        <Break />
        <Row>
            <Column>
                <Link href="#">FAQ</Link>
                <Link href="#">Investor Relations</Link>
                <Link href="#">Ways to Watch</Link>
                <Link href="#">Corporate Information</Link>
                <Link href="#">Netflix Originals</Link>
            </Column>

            <Column>
                <Link href="#">Help Centre</Link>
                <Link href="#">Jobs</Link>
                <Link href="#">Terms of Use</Link>
                <Link href="#">Contact Us</Link>
            </Column>

            <Column>
                <Link href="#">Account</Link>
                <Link href="#">Redeem gift cards</Link>
                <Link href="#">Privacy</Link>
                <Link href="#">Speed Test</Link>
            </Column>

            <Column>
                <Link href="#">Media Centre</Link>
                <Link href="#">Buy gift cards</Link>
                <Link href="#">Cookie Preferences</Link>
                <Link href="#">Legal Notices</Link>
            </Column>
        </Row>
        <Break />
        <Text>Netflix United Kingdom</Text>
    </Container>;
};

export default Footer;
