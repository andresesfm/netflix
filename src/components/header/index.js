import React, { useState } from 'react';
import { Container, Search, SearchInput, SearchIcon, Picture, Profile, Frame, Logo, ButtonLink, Dropdown, TextLink, Background, Feature, Text, FeatureCallout, Group, PlayButton } from './styles/header'
import { Link as ReactRouterLink } from 'react-router-dom';
import * as ROUTES from "../../constants/routes"
import logo from '../../logo.svg'
const Content = ({ children, ...restProps }) => <Container>
    <Frame {...restProps}>
        <HLogo to={ROUTES.SIGN_IN} src={logo} alt="Netflix" />
        <ButtonLink to={ROUTES.SIGN_IN} >Sign In</ButtonLink>
    </Frame>
    {children}
</Container>

const Header = ({ children, bg = true, ...restProps }) => {

    return (bg ?
        <Background {...restProps}>
            <Content >{children}</Content>
        </Background> :
        <Content >{children}</Content>);
};


const HLogo = ({ to, ...restProps }) => {
    return (<ReactRouterLink to={to}>
        <Logo {...restProps} />
    </ReactRouterLink>)
}

export default Header;

export const ProfileHeader = () => {
    return (
        <Frame>
            <HLogo to={ROUTES.HOME} src={logo} alt="Netflix" />
        </Frame>)
}

export const BrowseHeader = ({ children, user, signOut, searchTerm, setSearchTerm, category, setCategory, ...restProps }) => {
    return (
        <Background {...restProps}>
            <Frame>
                <Group>
                    <HLogo to={ROUTES.HOME} src={logo} alt="Netflix" />
                    <TextLink active={category === 'series'} onClick={() => setCategory('series')}>Series</TextLink>
                    <TextLink active={category === 'films'} onClick={() => setCategory('films')}>Films</TextLink>
                </Group>
                <Group>
                    <HSearch searchTerm={searchTerm} setSearchTerm={setSearchTerm} />
                    <Profile>
                        <Picture src={`/images/users/${user.photoURL}.png`} />
                        <Dropdown>
                            <Group>
                                <Picture src={`/images/users/${user.photoURL}.png`} />
                                <TextLink>{user.displayName}</TextLink>
                            </Group>
                            <Group>
                                <TextLink onClick={signOut}>Sign out</TextLink>
                            </Group>
                        </Dropdown>
                    </Profile>
                </Group>
            </Frame>
            <Feature>
                <FeatureCallout> Watch Joker now</FeatureCallout>
                <Text>Hello, this is a very good movie, enjoy</Text>
                <PlayButton >Play</PlayButton>
            </Feature>
        </Background>
    )
}

const HSearch = ({ children, searchTerm, setSearchTerm, ...restProps }) => {
    const [searchActive, setSearchActive] = useState(false);

    return <Search {...restProps}>
        <SearchIcon onClick={() => { setSearchActive(active => !active) }}>
            <img src='/images/icons/search.png' alt='Search' />
        </SearchIcon>
        <SearchInput value={searchTerm}
            onChange={({ target }) => { console.log('target', target.value); setSearchTerm(target.value) }}
            placeholder='Search films and series'
            active={searchActive} />
    </Search>;
};
