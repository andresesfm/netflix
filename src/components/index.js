export { default as Jumbotron } from './jumbotron'
export { default as Footer } from './footer'
export { default as Accordion } from './accordion'
export { default as Header } from './header'
export { default as Feature } from './feature'
export { default as SigninForm } from './form/signin'
export { default as SignupForm } from './form/signup'
export { default as SelectProfile } from './profile'
export { default as Loading } from './loading'
export { BrowseHeader } from './header'
export { default as Card } from './card'
export { default as Player } from './player'

