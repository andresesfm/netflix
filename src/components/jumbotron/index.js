import React from 'react';
import { Inner, Container, Item, Title, SubTitle, Image } from './styles/jumbotron';

const index = ({ children, direction = 'row', data, ...restProps }) => {
    return (
        <Container>
            {data.map(({ id, direction, title, image, alt, subTitle }) =>
            (<Item key={id}><Inner direction={direction} >
                <Title>{title}</Title>
                <SubTitle>{subTitle}</SubTitle>
                <Image src={image} alt={alt} />
            </Inner>
            </Item>))}
        </Container>)

};

export default index;
