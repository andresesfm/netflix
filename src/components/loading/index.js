import React from 'react';
import { Spinner, LockBody, Picture } from './styles/loading'

const Loading = ({ src, ...restProps }) => {
    return <Spinner {...restProps}>
        <LockBody />
        <Picture src={`/images/users/${src}.png`} />
    </Spinner>;
};

export default Loading;
