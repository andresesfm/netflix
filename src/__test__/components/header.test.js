import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Header, BrowseHeader, ProfileHeader } from '../../components';
import { MemoryRouter } from 'react-router-dom'


describe('Header is rendered', () => {
    it('renders Header as expected with data', () => {
        const { container } = render(
            <MemoryRouter>
                <Header category='catt' />
            </MemoryRouter>)

        expect(container.firstChild).toMatchSnapshot();
    })
    it('renders as expected with no bg', () => {
        const { container } = render(
            <MemoryRouter>
                <Header category='catt' bg={false} />
            </MemoryRouter>)
        expect(container.firstChild).toMatchSnapshot();
    })
    it('Renders BrowseHeader and toggles the search', () => {
        const user = { photoURL: '5', displayName: 'Andy' }
        const setCategory = jest.fn()
        const setSearchTerm = jest.fn()
        const { container, queryByText, getByAltText, getByPlaceholderText } = render(
            <MemoryRouter>
                <BrowseHeader
                    category='films'
                    user={user}
                    setCategory={setCategory}
                    setSearchTerm={setSearchTerm} />
            </MemoryRouter>)
        const films = queryByText('Films')
        expect(films).toBeTruthy()
        const series = queryByText('Series')
        expect(series).toBeTruthy()
        expect(series).not.toHaveAttribute('active')
        fireEvent.click(series)
        fireEvent.click(films)
        expect(setCategory).toBeCalledTimes(2)
        fireEvent.click(getByAltText('Search'))
        fireEvent.change(getByPlaceholderText('Search films and series'), { target: { value: 'Sponge' } })
        expect(setSearchTerm).toBeCalledTimes(1)

        expect(container.firstChild).toMatchSnapshot();

    })
    it('renders ProfileHeader as expected with data', () => {
        const { container, getByAltText } = render(
            <MemoryRouter>
                <ProfileHeader />
            </MemoryRouter>)
        expect(getByAltText('Netflix')).toBeTruthy()
        expect(container.firstChild).toMatchSnapshot();
    })
})