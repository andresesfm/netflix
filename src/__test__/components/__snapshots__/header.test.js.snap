// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header is rendered Renders BrowseHeader and toggles the search 1`] = `
<div
  class="sc-llYSUQ eMmjTr"
>
  <div
    class="sc-bBHxTw bnmefL"
  >
    <div
      class="sc-kLwhqv hjsoOO"
    >
      <a
        href="/"
      >
        <img
          alt="Netflix"
          class="sc-cxpSdN jfoCMi"
          src="logo.svg"
        />
      </a>
      <p
        class="sc-bYoBSM VoFXY"
      >
        Series
      </p>
      <p
        class="sc-bYoBSM lgdNok"
      >
        Films
      </p>
    </div>
    <div
      class="sc-kLwhqv hjsoOO"
    >
      <div
        class="sc-gWXbKe keGfLq"
      >
        <button
          class="sc-cCcXHH icyLFi"
        >
          <img
            alt="Search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="sc-cidDSM hVEivb"
          placeholder="Search films and series"
          value=""
        />
      </div>
      <div
        class="sc-hiCibw jkrfzC"
      >
        <button
          class="sc-ikJyIC ikPfQA"
          src="/images/users/5.png"
        />
        <div
          class="sc-jJoQJp iVgypj"
        >
          <div
            class="sc-kLwhqv hjsoOO"
          >
            <button
              class="sc-ikJyIC ikPfQA"
              src="/images/users/5.png"
            />
            <p
              class="sc-bYoBSM VoFXY"
            >
              Andy
            </p>
          </div>
          <div
            class="sc-kLwhqv hjsoOO"
          >
            <p
              class="sc-bYoBSM VoFXY"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-fKVqWL sc-iJKOTD hZGtji gtkals"
  >
    <h2
      class="sc-giYglK dKWYEW"
    >
       Watch Joker now
    </h2>
    <p
      class="sc-ezbkAF dZyvsg"
    >
      Hello, this is a very good movie, enjoy
    </p>
    <button
      class="sc-jcFjpl kAwcdw"
    >
      Play
    </button>
  </div>
</div>
`;

exports[`Header is rendered renders Header as expected with data 1`] = `
<div
  class="sc-llYSUQ eMmjTr"
>
  <div
    class="sc-fKVqWL hZGtji"
  >
    <div
      class="sc-bBHxTw bnmefL"
    >
      <a
        href="/signin"
      >
        <img
          alt="Netflix"
          class="sc-cxpSdN jfoCMi"
          src="logo.svg"
        />
      </a>
      <a
        class="sc-iwjdpV grsJjB"
        href="/signin"
      >
        Sign In
      </a>
    </div>
  </div>
</div>
`;

exports[`Header is rendered renders ProfileHeader as expected with data 1`] = `
<div
  class="sc-bBHxTw bnmefL"
>
  <a
    href="/"
  >
    <img
      alt="Netflix"
      class="sc-cxpSdN jfoCMi"
      src="logo.svg"
    />
  </a>
</div>
`;

exports[`Header is rendered renders as expected with no bg 1`] = `
<div
  class="sc-fKVqWL hZGtji"
>
  <div
    class="sc-bBHxTw bnmefL"
  >
    <a
      href="/signin"
    >
      <img
        alt="Netflix"
        class="sc-cxpSdN jfoCMi"
        src="logo.svg"
      />
    </a>
    <a
      class="sc-iwjdpV grsJjB"
      href="/signin"
    >
      Sign In
    </a>
  </div>
</div>
`;
